fem/restriction.cpp

```diff
@@ -152,24 +152,18 @@ void ElementRestriction::Mult(const Vector& x, Vector& y) const
    const int nd = dof;
    const int vd = vdim;
    const bool t = byvdim;
-   auto d_offsets = offsets.Read();
-   auto d_indices = indices.Read();
    auto d_x = Reshape(x.Read(), t?vd:ndofs, t?ndofs:vd);
    auto d_y = Reshape(y.Write(), nd, vd, ne);
-
-   MFEM_FORALL(i, ndofs,
+   auto d_gatherMap = gatherMap.Read();
+   MFEM_FORALL(i, dof*ne,
    {
-      const int offset = d_offsets[i];
-      const int nextOffset = d_offsets[i+1];
+      const int gid = d_gatherMap[i];
+      const bool plus = gid >= 0;
+      const int j = plus ? gid : -1-gid;
       for (int c = 0; c < vd; ++c)
       {
-         const double dofValue = d_x(t?c:i,t?i:c);
-         for (int j = offset; j < nextOffset; ++j)
-         {
-            const bool positive = (d_indices[j] >= 0);  // otherwise, sign must be flipped
-            const int idx_j = positive ? d_indices[j] : -1 - d_indices[j];
-            d_y(idx_j % nd, c, idx_j / nd) = positive ? dofValue : -dofValue;
-         }
+         const double dofValue = d_x(t?c:j, t?j:c);
+         d_y(i % nd, c, i / nd) = plus ? dofValue : -dofValue;
       }
```

Comment:
Is the logic here still the same?

Reply:
We had to solve the conflict, but it should be fine.